tests/nexus_workflow_test.go

```diff
@@ -689,9 +689,35 @@ func (s *NexusWorkflowTestSuite) TestNexusOperationAsyncCompletion() {
 		s.NoError(err)
 		if event.EventType == enumspb.EVENT_TYPE_NEXUS_OPERATION_COMPLETED {
 			seenCompletedEvent = true
+			break
 		}
 	}
 	s.True(seenCompletedEvent)
+
+	// Reset the workflow again to the same point with enumspb.RESET_REAPPLY_EXCLUDE_TYPE_NEXUS option
```

Comment:
Should we test both started and completed can be reset?

Reply:
Unfortunately, it's a bit difficult to test the re-application of the started event in this setup because there is no valid reset point between nexus events scheduled & started. In the test they are consecutive. 
So I tried to reset to a point before nexus scheduled event and that's when I discovered the [issue I posted above](https://github.com/temporalio/temporal/pull/6598/files#r1792855547).

Reply:
Ah I see. we could craft the scenario to generate an extra task before the started event but I think this is good enough.